include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_kernels.h

```diff
@@ -570,12 +570,23 @@ struct __radix_sort_onesweep_kernel
                                                __local_tid * __bin_width * sizeof(_LocOffsetT),
                                            __scan<_LocOffsetT, _LocOffsetT>(__thread_grf_hist_summary));
 
-            // 1.3 Copy the histogram at the region designited for synchronization between work-groups.
+            // 1.3. Copy the histogram at the region designated for synchronization between work-groups.
             // Set the status as "updated".
             if (__wg_id != 0)
-                __dpl_esimd::__block_store<::std::uint32_t, __bin_width>(
+            {
+                // Write the histogram explicitly to L2, bypassing L1.
+                __dpl_esimd::__ens::lsc_block_store<::std::uint32_t, __bin_width,
+                                                    __dpl_esimd::__ens::lsc_data_size::default_size,
+                                                    __dpl_esimd::__ens::cache_hint::uncached,
+                                                    __dpl_esimd::__ens::cache_hint::write_back>(
                     __p_this_group_hist + __local_tid * __bin_width, __thread_grf_hist_summary | __hist_updated);
+            }
         }
+        // Make sure the histogram updated at the step 1.3 is visible to other groups
+        // The histogram data above is in L2, which is assumed to be coherent: no need to flush it
+        __dpl_esimd::__ns::fence<__dpl_esimd::__ns::memory_kind::global,
+                                 __dpl_esimd::__ns::fence_flush_op::none,
```

Comment:
The documentation for `cache_hint::write_back` states that the data will only be written to the furthest cache upon eviction. However, `fence_flush_op::none` does not actually evict any lines. Should this be `fence_flush_op::evict`? Or should the previous `lsc_block_store` use `cache_hint::write_through` instead, which would "immediately write data to the subsequent furthest cache" and not require eviction for it to be visible?

Reply:
Thanks for pointing it out. I initially thought it is enough to leave the data in L2 as all work-groups "sit" on the same L2. However, there may be multiple stacks, were it is not true.

> Should this be fence_flush_op::evict?

It is one of the options. However, I would prefer `fence_flush_op::clean`, because it does not erase the cache.

> Or should the previous lsc_block_store use cache_hint::write_through instead, which would "immediately write data to the subsequent furthest cache" and not require eviction for it to be visible?

It would be perfect to leave the data in L2 and write to the global memory, what `cache_hint::write_through` could do, but this option is not available, see [store functions: valid cache hint combinations](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md#valid-combinations-of-l1-and-l2-cache-hints-for-store-functions). Alternative is to use `cache_hint::uncached`, which, I suppose, writes directly to the global memory, forcing the consumer going further, bypassing L2.

I did performance checks and `cache_hint::uncached` is a better choice: ~3% slow down on average, compared to 6% for `fence_flush_op::clean`, and 7% for `fence_flush_op::evict`.